docs/source/models.rst

```diff
@@ -182,11 +189,172 @@ TensorFlow (``tensorflow``)
+
+        def predict(self, context, model_input):
+            input_matrix = xgb.DMatrix(model_input.values)
+            return self.xgb_model.predict(input_matrix)
+
+    # Create a Conda environment for the new MLflow model that contains the XGBoost library
+    # as a dependency, as well as the required CloudPickle library
+    import cloudpickle
+    conda_env = {
+        'channels': ['defaults'],
+        'dependencies': [
+          'xgboost={}'.format(xgb.__version__)
+          'cloudpickle={}'.format(cloudpickle.__version__)
+        ],
+        'name': 'xgb_env'
+    }
+
+    # Save the MLflow model
+    mlflow_pyfunc_model_path = "xgb_mlflow_pyfunc"
+    mlflow.pyfunc.save_model(
+            dst_path=mlflow_pyfunc_model_path, python_model=XGBWrapper(), artifacts=artifacts,
+            conda_env=conda_env)
+
+    # Load the model in `python_function` format
+    loaded_model = mlflow.pyfunc.load_pyfunc(mlflow_pyfunc_model_path)
+
+    # Evaluate the model
+    import pandas as pd
+    test_predictions = loaded_model.predict(pd.DataFrame(x_test))
+    print(test_predictions)
+
+.. _custom-flavors:
+
 Custom Flavors
---------------
-You can add a flavor in MLmodel files, either by writing it directly or
-building it with the :py:class:`mlflow.models.Model` class. Choose an arbitrary string name
-for your flavor. MLflow tools ignore flavors in the MLmodel file that they do not understand.
+^^^^^^^^^^^^^^
+You can also create custom MLflow models by writing a custom *flavor*.
+
+As discussed in the :ref:`model-api` and :ref:`model-storage-format` sections, an MLflow model
+is defined by a directory of files that contains an ``MLmodel`` configuration file. This ``MLmodel``
+file describes various model attributes, including the flavors in which the model can be 
+interpreted. The ``MLmodel`` file contains an entry for each flavor name; each entry is
+a YAML-formatted collection of flavor-specific attributes.
+
+In order to create a new flavor to support a custom model, you simply need to define the set
+of flavor-specific attributes to include in the ``MLmodel`` configuration file, as well as the
+code that can interpret the contents of the model directory and the flavor's attributes.
+
+As an example, let's examine the :py:mod:`mlflow.pytorch` module corresponding to MLflow's 
+``pytorch`` flavor. In the :py:func:`mlflow.pytorch.save_model()` method, a PyTorch model is saved 
+to a specified output directory. Additionally, :py:func:`mlflow.pytorch.save_model()` leverages the 
+:py:func:`mlflow.models.Model.add_flavor()` and :py:func:`mlflow.models.Model.save()` functions to 
+produce an ``MLmodel`` configuration containing the ``pytorch`` flavor. The resulting configuration 
+has several flavor-specific attributes, such as ``pytorch_version``, which denotes the version of the 
+PyTorch library that was used to train the model. In order to interpret model directories
```

Comment:
In order to interpret model directories -> To interpret model directories

Reply:
Done!